### PR TITLE
LibCore+Userland: Teach file utility about 13 new formats

### PR DESCRIPTION
**Pull request by vkoskiv**
I added 12 new formats, as well as a new `zip_details` description generator to `file` that uses the facilities in `LibArchive` to enumerate ZIP files.
While doing this, I noted that LibArchive fails to report some directories, as shown in this picture here below

image

I believe there are at least 2 directories here. `dir/` and `dir/empty/`, but this is beyond the scope of this PR. (This zip file can be found in `Toolchain/Tarballs/gcc-12.1.0/libgo/go/archive/zip/testdata/`)

While working on adding new formats to `file`, I sorted the enumerations in `LibCore/MimeData.cpp` alphabetically, which improved the readability considerably, and made it much easier to match them to the corresponding descriptions in `file`.

### CODE REVIEW

**Comment by vkoskiv** [comment]
@LucasChollet Agreed! Had that exact same idea while I was working on this. The current API in `LibCore` is a bit awkward:

```
auto file_name_guess = Core::guess_mime_type_based_on_filename(path);
auto mime_type = Core::guess_mime_type_based_on_sniffed_bytes(*file).value_or(file_name_guess);
```

Would be nice to have, something like

```
ErrorOr<Core::MimeType> mime_type = Core::guess_mime_type(path);
```

And just have it use whatever trickery it needs in there to come up with the best possible guess, and an error if nothing came up.

I still think `file.cpp` is the best place for the mime-type -> human readable description mappings. Perhaps the simple descriptions could be baked into `Core::MimeType`, but the specialized descriptions are beyond the scope of the `LibCore` functionality, I think.

**Comment by LucasChollet** [comment]
Also, make sure to not use `‑` (Non-breaking Hyphen U+2011) inside mime types.
It's currently the case for the three `x-portable*`  in the "guess from filename" list and it should't.

**Comment by AtkinsSJ** [comment]
I agree with the discussion here, combining the file-identification stuff together would be very comfy. Are you wanting to modify this PR to work that way, or shall I merge this now and you can make the adjustments in a future PR? I think it's good to go otherwise.

**Comment by vkoskiv** [comment]
@LucasChollet Good catch! How did you notice that?

Looks like those symbols have been there all the way since 2020 when this code was in LibWeb, introduced in 41d5531, 0f7a651 and 97aca8f in #2611 

@AtkinsSJ I just pushed a fix for the above, you can merge this now.

File: Userland/Libraries/LibCore/MimeData.cpp [hunk 1]
```diff
@@ -54,56 +54,45 @@ void MimeData::set_text(DeprecatedString const& text)
     set_data("text/plain", text.to_byte_buffer());
 }
 
+static AK::Vector<StringView> s_plaintext_suffixes = {
```
Comment (LucasChollet) [inline Userland/Libraries/LibCore/MimeData.cpp:57]: Small nit, both of these can be Array instead of Vector

**Review by LucasChollet — COMMENTED** [review]
Just a small nit, otherwise it's good to go

**Comment by LucasChollet** [comment]
> How did you notice that?

During a ~~fun~~ debugging session
`file` currently doesn't work on portable format files because of that (and also because only ASCII variants are registered for byte sniffing, but I'll push a commit for that) 

Do you want to tackle the refactoring or should I go for it @vkoskiv?

File: Userland/Libraries/LibCore/MimeData.cpp (hunk 1 above)
Comment (vkoskiv) [reply Userland/Libraries/LibCore/MimeData.cpp]: Fixed

**Comment by vkoskiv** [comment]
@LucasChollet You can go ahead with the refactor

**Pull request merged**